src/fairchem/core/modules/scaling/fit.py

```diff
@@ -31,7 +31,9 @@ def _prefilled_input(prompt: str, prefill: str = "") -> str:
 
 def _train_batch(trainer: BaseTrainer, batch) -> None:
```

Comment:
decorator here too?

```python
@torch.no_grad
def _train_batch(trainer: BaseTrainer, batch) -> None:
```

Reply:
And cant we have the loss computation under autocast, too? if so then double up on decorators and add the torch.autocast as a decorator too.

Reply:
I dont think the decorator will work? can the decorator take a conditional or optional enabled variable, enabled=trainer.scaler is not None?